bin/commands/init/certificate/index.ts

```diff
@@ -0,0 +1,531 @@
+/*
+  This program and the accompanying materials are made available
+  under the terms of the Eclipse Public License v2.0 which
+  accompanies this distribution, and is available at
+  https://www.eclipse.org/legal/epl-v20.html
+ 
+  SPDX-License-Identifier: EPL-2.0
+ 
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as std from 'std';
+import * as common from '../../../libs/common';
+import * as stringlib from '../../../libs/string';
+import * as jsonlib from '../../../libs/json';
+import * as shell from '../../../libs/shell';
+import * as config from '../../../libs/config';
+
+const ZWE_CLI_PARM_KEYS = Object.keys(std.getenviron()).filter((key: string)=> {
+  return key.startsWith('ZWE_CLI_PARAMETER');
+});
+const ZWE_CLI_ENVS = {};
+ZWE_CLI_PARM_KEYS.forEach((key: string)=>{
+  ZWE_CLI_ENVS[key] = std.getenv(key);
+});
+
+function zweExec(command: string): void {
+  const result = shell.execZweSync(command, ZWE_CLI_ENVS);
+  if (result.rc != 0) {
+    common.printErrorAndExit(`Error ZWEL0305E: Failed to call certificate command, rc=${result.rc}.`, undefined, 305);
+  }
+}
+
+export function execute() {
+  common.printLevel1Message(`Initializing Zowe keystore`);
+
+  // Constants
+  // This is made because there's some dynamic var name assignment going on below,
+  // So we keep it all in an object rather than doing evals into a global
+  const CERT_PARMS: any = {};
+
+  // Validation
+  common.requireZoweYaml();
+  const zoweConfig = config.getZoweConfig();
+
+  // read prefix and validate
+  const prefix=zoweConfig.zowe?.setup?.dataset?.prefix;
+  if (!prefix) {
+    common.printErrorAndExit(`Error ZWEL0157E: Zowe dataset prefix (zowe.setup.dataset.prefix) is not defined in Zowe YAML configuration file.`, undefined, 157);
+  }
+  // read JCL library and validate
+  const jcllib=zoweConfig.zowe.setup.dataset.jcllib;
+  if (!jcllib) {
+    common.printErrorAndExit(`Error ZWEL0157E: Zowe custom JCL library (zowe.setup.dataset.jcllib) is not defined in Zowe YAML configuration file.`, undefined, 157);
+  }
+  let securityProduct=zoweConfig.zowe.setup.security?.product;
+  let securityUsersZowe=zoweConfig.zowe.setup.security?.users?.zowe;
+  let securityGroupsAdmin=zoweConfig.zowe.setup.security?.groups?.admin;
+  // read cert type and validate
+  const certType=zoweConfig.zowe.setup.certificate?.type;
+  if (!certType) {
+    common.printErrorAndExit(`Error ZWEL0157E: Certificate type (zowe.setup.certificate.type) is not defined in Zowe YAML configuration file.`, undefined, 157);
+  }
+  if (certType != "PKCS12" && certType != "JCERACFKS") {
```

Comment:
Is this up to date? 
https://github.com/zowe/zowe-install-packaging/blob/610827598b1c3deeb3a8467ab2745c1a0b04edef/example-zowe.yaml#L100

Reply:
no, this is months old and needs rebasing.